pkg/config/config_test.go

```diff
@@ -44,3 +45,28 @@ func TestNetworkBlockDevice_NoDevice(t *testing.T) {
 	nbdItem := vm.NetworkBlockDevice("nbd2")
 	require.Nil(t, nbdItem)
 }
+
+func TestVirtualMachine_ValidBlkDevices(t *testing.T) {
+	vm := &VirtualMachine{}
+
+	file, err := os.CreateTemp("", "test-*.img")
+	require.NoError(t, err)
+	defer os.Remove(file.Name())
+
+	// write qcow2 magic number
+	_, err = file.Write([]byte("QFI\xfb"))
```

Comment:
This is not a good way to test. We duplicate the knowledge in the test, if our assumption is incorrect or incomplete the test will pass when the actual code is failing to detect the image.

The best way to ensure that we detect qcow2 images is to create a qcow2 image with tool like qemu-img and verify that we detect the image.